### PR TITLE
Csexp_rpc - Log to _build/log

### PR DESCRIPTION
**Pull request by rgrinberg**
Should be easier to debug than dumping to stderr

### CODE REVIEW

File: src/csexp_rpc/csexp_rpc.ml [hunk 1]
```diff
@@ -228,8 +228,20 @@ module Server = struct
     let loop () =
       let* accept = accept () in
       match accept with
-      | Error _
+      | Error `Stopped ->
+        Log.info [ Pp.text "RPC stopped accepting." ];
+        Fiber.return None
+      | Error (`Exn exn) ->
+        Log.info
```
Comment (aalekseyev) [inline src/csexp_rpc/csexp_rpc.ml:235]: Maybe `Log.error` here? (if that's even a thing)

Comment (rgrinberg) [reply src/csexp_rpc/csexp_rpc.ml:235]: It's not a thing unfortunately.

Comment (jeremiedimino) [reply src/csexp_rpc/csexp_rpc.ml:235]: Why not crash the process completely here? Having a dune server that continues running but doesn't accept connections seems ... not great?

Comment (cwong-ocaml) [reply src/csexp_rpc/csexp_rpc.ml:235]: If there's a pending build, I can see a good case for completing the build and *then* shutting down.

**Comment by cwong-ocaml** [comment]
No immediate objections. I'm torn between whether I think this will lead to a too-cluttered log for more complex interactions, but we can deal with that if/when it becomes an issue.

File: src/csexp_rpc/csexp_rpc.ml (hunk 1 above)
Comment (jeremiedimino) [reply src/csexp_rpc/csexp_rpc.ml:235]: On the other hand, if it's an half an hour build the user might be wondering why the key bindings they are typing in their editor have no effect.

In any, accept errors might be rare enough that this is low-priority.

Comment (rgrinberg) [reply src/csexp_rpc/csexp_rpc.ml:235]: It also seems bad to immediately terminate while there are still active sessions. But if you prefer this behavior, I can do that.

**Comment by cwong-ocaml** [comment]
Will this cause problems where `dune rpc init` and `dune build -w` will clobber each other's logs?

**Comment by rgrinberg** [comment]
> Will this cause problems where dune rpc init and dune build -w will clobber each other's logs?

Good point. I will change the log file path for rpc init.

**Comment by cwong-ocaml** [comment]
This isn't the PR to address it, but I believe this is also an issue with the various other dune rpc commands (shutdown, `rpc build`, etc)

**Comment by rgrinberg** [comment]
> Would be nice to get this merged sooner rather than later.

Didn't we decide that it's wrong for rpc clients to log to _build/log?

We could fix this, but we could also just get rid of `dune rpc init` as
we planned and the problem will go away. Without dune rpc init, only the
server will use _build/log to log.

**Comment by cwong-ocaml** [comment]
> Didn't we decide that it's wrong for rpc clients to log to _build/log?

That's true, my mistake.

**Comment by rgrinberg** [comment]
Now the logging is done only on the server side.

**Pull request merged**